libs/datamodel/core/src/ast/reformat/reformatter.rs

```diff
@@ -362,9 +383,65 @@ impl<'a> Reformatter<'a> {
         }
     }
 
+    fn extract_and_sort_directives<'i>(token: &'i Token, is_field_directive: bool) -> Vec<Pair<'i, Rule>> {
+        // get indices of directives and store in separate Vector
+        let mut directive_indices = Vec::new();
+        let mut directives = Vec::new();
+        for (i, pair) in token.clone().into_inner().enumerate() {
+            match pair.as_rule() {
+                Rule::directive => {
+                    directive_indices.push(i);
+                    directives.push(pair);
+                }
+                _ => (),
+            }
+        }
+
+        // sort directives
+        let correct_order = if is_field_directive {
+            vec!["map", "id", "unique", "default", "relation", "updatedAt"]
+        } else {
+            vec!["map", "unique", "index", "id"]
+        };
+        directives.sort_by(|a, b| {
+            let mut sort_index_a = usize::MAX;
+            let mut sort_index_b = usize::MAX;
+            for (i, name) in correct_order.iter().enumerate() {
+                if a.as_str().contains(name) {
+                    sort_index_a = i;
+                    break;
+                }
+            }
+            for (i, name) in correct_order.iter().enumerate() {
+                if b.as_str().contains(name) {
+                    sort_index_b = i;
+                    break;
+                }
+            }
+            sort_index_a.cmp(&sort_index_b)
+        });
+
+        return directives;
+    }
+
     fn reformat_field(&self, target: &mut TableFormat, token: &Token, model_name: &str) {
         let field_name = &Self::get_identifier(token);
-        for current in token.clone().into_inner() {
+
+        // extract and sort directives
+        let directives = Self::extract_and_sort_directives(token, true);
+
+        // iterate through tokens and reorder directives
+        let mut count = 0;
+        let sorted_inner_pairs = token.clone().into_inner().map(|p| match p.as_rule() {
```

Comment:
It might be good to mention what has been sorted. Something to make clear that only the order of directives has changed.